filter/filter.go

```diff
@@ -2,19 +2,22 @@ package filter
 
 import (
 	"errors"
-	"fmt"
 	"strings"
 
 	"github.com/brimsec/zq/ast"
 	"github.com/brimsec/zq/expr"
 	"github.com/brimsec/zq/pkg/byteconv"
 	"github.com/brimsec/zq/zcode"
 	"github.com/brimsec/zq/zng"
-	"github.com/brimsec/zq/zng/resolver"
 )
 
 type Filter func(*zng.Record) bool
 
+type Program interface {
+	AsFilter() (Filter, error)
+	AsBufferFilter() (*BufferFilter, error)
+}
```

Comment:
I think this ought to live in zbuf/scanner.go since its use there motivates its existence.

Reply:
It felt lonely in package filter.  Since it's in zbuf, I called it zbuf.Filter.